### PR TITLE
Update run_tests to generate xml reports

### PR DESCRIPTION
**Pull request by nitsakh**
Add ability to generate xml report for test results.

### CODE REVIEW

File: script/run_tests [hunk 1]
```diff
@@ -51,6 +51,9 @@ def RunTests(binary_tests):
       if disabled_tests and test_name in disabled_tests and disabled_tests[test_name]:
         exclusion_filter = '-' + ':'.join(disabled_tests[test_name])
         test_args.append('--gtest_filter=%s' % exclusion_filter)
+      # Add reporting path if enabled
+      if generate_report:
+        test_args.append("--gtest_output=xml:" + os.path.join(SOURCE_ROOT, 'test_reports', test_name + os.path.extsep + 'xml'))
```
[the remark this answers is not in the document]
Comment (jkleinsc) [reply script/run_tests]: @alexeykuzmin I think we can parse multiple files.  Location shouldn't matter too much.  I am going to update this PR with the appropriate CI configs.

Comment (alexeykuzmin) [inline script/run_tests]:
```python
os.path.join(SOURCE_ROOT, 'test_reports', test_name + os.path.extsep + 'xml')
```
@nitsakh Can you please move it to a function? Looks really complicated inline )

**Comment by jkleinsc** [comment]
FYI @nitsakh @alexeykuzmin:
Results in CircleCI:
https://circleci.com/gh/electron/libchromiumcontent/6405

**Comment by jkleinsc** [comment]
Also FYI @nitsakh and @alexeykuzmin our Mac and Windows CIs didn't have the yaml module installed on them, so they were throwing this error:
```Traceback (most recent call last):
  File "script\run_tests", line 9, in <module>
    import yaml
ImportError: No module named yaml
```
I resolved this on Windows by running `pip install pyyaml`.  I need to do the same for our Mac build agents.

**Comment by alexeykuzmin** [comment]
@jkleinsc 
We have pyyaml as a submodule.
In order to make `import yaml` work we only should do this:

```python
from config import VENDOR_DIR

PYYAML_LIB_DIR = os.path.join(VENDOR_DIR, 'pyyaml', 'lib')
sys.path.append(PYYAML_LIB_DIR)
import yaml
```

**Comment by alexeykuzmin** [comment]
> https://circleci.com/gh/electron/libchromiumcontent/6405
> "2987 failures"
That's _slightly_ more than I expected ) Do all of these tests fail locally?
@nitsakh

**Comment by jkleinsc** [comment]
@alexeykuzmin in regards to the failures, I think its because we are running headless and xvfb wasn't running.  I'll update to run with xvfb.

File: script/run_tests (hunk 1 above)
Comment (nitsakh) [reply script/run_tests]: It doesn't do much. I can change `os.path.extsep + 'xml'` to directly `.xml` is that'll make it simpler. Extension separators are same on Win, Mac and Linux anyway.

Comment (alexeykuzmin) [reply script/run_tests]: The problem is not that it's doing to much.
It's that I have to read it through to understand what's going on there.
Having `test_args.append("--gtest_output=xml:" + get_output_file_path(test_name))` is much easier to read.

Yeah, I'm kinda nitpicking here, sorry =/

**Comment by jkleinsc** [comment]
Closing in favor of #530 which continues this PR.

**Pull request closed without merging**